BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -8,28 +8,36 @@ import Common
 // MARK: Protocol
 public protocol TabDataStore {
     func fetchWindowData() async -> WindowData
-    func saveWindowData(window: WindowData) async
+    func saveWindowDataWithBackup(window: WindowData) async
     func clearAllWindowsData() async
     func fetchWindowData(withID id: UUID) async -> WindowData?
     func fetchAllWindowsData() async -> [WindowData]
     func clearWindowData(for id: UUID) async
+    func fetchBackupWindowData(forID id: UUID) async -> WindowData?
```

Comment:
This method should also not be exposed and the restoration logic should be handled within the fetchWindowData method.